services/semaphore.js

```diff
@@ -7,20 +8,41 @@ module.exports = {
 		return Boolean(env.SEMAPHORE);
 	},
 	configuration({env, cwd}) {
-		const pr = env.PULL_REQUEST_NUMBER;
-		const isPr = Boolean(pr);
+		if (env.BRANCH_NAME) {
+			// Exists in v1 only
+			const pr = env.PULL_REQUEST_NUMBER;
+			const isPr = Boolean(pr);
+			return {
+				name: 'Semaphore',
+				service: 'semaphore',
+				commit: head({env, cwd}),
+				build: env.SEMAPHORE_BUILD_NUMBER,
+				branch: isPr ? undefined : env.BRANCH_NAME,
+				pr,
+				isPr,
+				prBranch: isPr ? env.BRANCH_NAME : undefined,
+				slug: env.SEMAPHORE_REPO_SLUG,
+				root: env.SEMAPHORE_PROJECT_DIR,
+			};
+		}
 
+		let parsedRepo;
+
+		try {
```

Comment:
Why doing that instead of using `env.SEMAPHORE_REPO_SLUG`?

Reply:
In v2, there is no `SEMAPHORE_REPO_SLUG`. The remaining way to identify `owner/project` is to parse the repository URL. As mentionned below, I will set `slug:undefined` for semaphore v2.